contracts/core/Dispatcher.sol

```diff
@@ -65,18 +71,27 @@ contract Dispatcher is IbcDispatcher, IbcEventsEmitter, Ownable, Ibc {
         bytes memory strBytes = bytes(hexStr);
         bytes memory addrBytes = new bytes(20);
 
-        for (uint256 i = 0; i < 20; i++) {
-            uint8 high = uint8(strBytes[i * 2]);
-            uint8 low = uint8(strBytes[1 + i * 2]);
-            // Convert to lowercase if the character is in uppercase
-            if (high >= 65 && high <= 90) {
-                high += 32;
-            }
-            if (low >= 65 && low <= 90) {
-                low += 32;
+
+        for (uint256 i = 0; i < 20;) {
```

Comment:
I think these bounds are all correct, but in this specific instance, I'm also slightly leaning towards "this unchecked math is too complicated for me to be absolutely sure that there isn't an over/underflow risk". 

Especially for L2s, I think the gas savings here would be negligible compared to the additional under/overflow risk added here. 

But also, I want to at some point take a stab at replacing this method completely with the `toHexStr` method in the OZ's strings library as well. Though it seemed like @nicopernas had imported that library in some parts of this codebase so maybe he has already tried that and it's not useful here.